### ISSUE TITLE
"cannot be converted to sql” error with custom function

### ISSUE THREAD

**Issue by trs79** [linq2db/linq2db#353]
I have a custom server-side only function that is very similar to the [MySum](https://github.com/linq2db/linq2db/blob/master/Tests/Linq/Linq/FunctionTests.cs#L465) function in the Linq2db unit tests:

```cs
[Sql.Function(Name = "UniqueValue", ServerSideOnly = true)]
public static int? UniqueValue( Expression<Func<ModelClass, string>> arg) {
            throw new NotSupportedException();
}
```

I then try to use it in a linq query with grouping:

```cs
var query = from a in context.data
                    group a by a.field1 into newgroup
                    select new {final = UniqueValue(row => row.field2) };
```

I then get "row => row.field2' cannot be converted to SQL." error at runtime. Is there another way to get a custom aggregate function to be called?

**Comment by sdanyliv** [comment]
@ili, i think we can extend AggregationBuilder to handle special attribute for such kind of functions.
What do you think?

**Comment by ili** [comment]
In general this sample should work

**Comment by ili** [comment]
Fixed in #775

**Comment by sdanyliv** [comment]
Sample code can be found here https://github.com/linq2db/linq2db/blob/master/Tests/Linq/Linq/FunctionTests.cs
Test `CustomAggregate`
